net/src/main/java/io/bitcoinsv/jcl/net/protocol/handlers/blacklist/BlacklistTracker.java

```diff
@@ -0,0 +1,90 @@
+package io.bitcoinsv.jcl.net.protocol.handlers.blacklist;
+
+import io.bitcoinsv.jcl.net.network.events.PeersBlacklistedEvent;
+
+import java.net.InetAddress;
+import java.time.Duration;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Copyright (c) 2024 nChain Ltd
+ *
+ * Used to track reasons & times/counts peers were blacklisted with. Can also be used to obtain duration for blacklist
+ * reasons from the config
+ *
+ * @author nChain Ltd
+ */
+public class BlacklistTracker {
+
+    private final Map<InetAddress, Map<PeersBlacklistedEvent.BlacklistReason, Long>> blacklistReasonCounters = new ConcurrentHashMap<>();
+    private final BlacklistHandlerConfig config;
+
+    private BlacklistTracker() { //hidden
+        this(null);
+    }
+
+    public BlacklistTracker(BlacklistHandlerConfig config) {
+        this.config = config;
+    }
+
+    public Long findExistingReasonCount(InetAddress inetAddress, PeersBlacklistedEvent.BlacklistReason reason) {
+        return blacklistReasonCounters.computeIfAbsent(inetAddress, key -> new ConcurrentHashMap<>()) //get or create a new reason-counter map for given IP address
+            .get(reason);
+    }
+
+    /**
+     * Inspects {@link #config}'s {@link io.bitcoinsv.jcl.net.protocol.handlers.blacklist.BlacklistHandlerConfig.BlacklistEntryConfiguration}s
+     * to find duration configured for the given reason and number of times the peer has been blacklisted.<br>
+     * If no configuration is provided, the default values ({@link BlacklistHandlerConfig#DEFAULT_BLACKLIST_ENTRY_CONFIGURATIONS}) will be used.
+     *
+     * @param reason           {@link PeersBlacklistedEvent.BlacklistReason} to match
+     * @param timesBlacklisted Number of times the peer has been blacklisted
+     * @return Configured {@link Duration} if matched for the given parameters, otherwise an empty {@link Optional}
+     */
+    public Optional<Duration> findConfiguredDuration(PeersBlacklistedEvent.BlacklistReason reason, Long timesBlacklisted) {
+        var counter = timesBlacklisted != null ? timesBlacklisted : 1L;
+        var configurationsToCheck = BlacklistHandlerConfig.DEFAULT_BLACKLIST_ENTRY_CONFIGURATIONS;
```

Comment:
accessing the DEFAULT list of Blacklist Configs should not be needed, only config.getBlacklistEntryConfigurations().
The constructors in BlacklistHandlerConfig already set the List to default if not specified, so no need to do this check here. 
If you are trying to rely to Default Values if a NULL List of BlacklistEntries are sent to the Constructor, a better place to do this in the Constructor itself:

```
public BlacklistHandlerConfig(ProtocolBasicConfig basicConfig, List<BlacklistEntryConfiguration> blacklistEntryConfigurations) {
        this.basicConfig = basicConfig;
        this.blacklistEntryConfigurations = (blacklistEntryConfigurations != null)
          ? blacklistEntryConfigurations
          : DEFAULT_BLACKLIST_ENTRY_CONFIGURATIONS);
    }
```